packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2356,3 +2362,47 @@ const waitForSeq = async (
     };
     deltaManager.on("op", handleOp);
 });
+
+export interface ISummarizerRequestOptions {
+    cache: boolean,
+    reconnect: boolean,
+    summarizingClient: boolean,
+}
+
+/**
+ * Forms a function that will request a Summarizer.
+ * @param loaderRouter - the loader acting as an IFluidRouter
+ * @param lastSequenceNumber - the last sequence number (e.g., from DeltaManager)
+ * @param cache - use cache to retrieve summarizer
+ * @param summarizingClient - is summarizer client
+ * @param reconnect - can reconnect on connection loss
+ */
+export const formRequestSummarizerFn = (
+    loaderRouter: IFluidRouter,
+    lastSequenceNumber: number,
+    { cache, reconnect, summarizingClient }: ISummarizerRequestOptions,
+) => async () => {
+    // TODO eventually we may wish to spawn an execution context from which to run this
```

Comment:
I do not think such TODOs are helpful in the code - they get stale the moment we check in code like that.
If you want to track it, I'd open an issue.

Reply:
This comment came with some code that I refactored. I'll remove it.